docs/user-tutorial/data-diagnosis.md

```diff
@@ -0,0 +1,142 @@
+---
+id: data-diagnosis
+---
+
+# Data Diagnosis
+
+## Introduction
+
+This tool is to filter the defective machines from thousands of benchmarking results automatically according to rules defined in **Rule file**.
+
+## Input
+
+The input mainly includes 2 files:
+
+ - **raw data**: jsonl file including multiple nodes' results automatically generated by SuperBench runner.
+
+    `Tips: the file can be found at ${output-dir}/results-summary.jsonl after each successful run.`
+
+ - **Rule file**: It uses YAML format and includes each metrics' rules to filter defective servers for diagnosis.
+
+- **baseline file**: json file including the baseline values for the metrics defined in the Rule file.
+
+### Rule file
+
+This section describes how to write rules in **Rule file**.
+
+The convention is the same with [SuperBench Config File](https://microsoft.github.io/superbenchmark/docs/superbench-config), please view it first.
+
+Here is an overview of the Rule file structure:
+
+scheme:
+```yaml
+version: string
+superbench:
+  var:
+    ${var_name}: dict
+  rules:
+    ${rule_name}:
+      function: string
+      criteria: string
+      categories: string
+      metrics:
+        - ${benchmark_name}/regex
+        - ${benchmark_name}/regex
+        ...
+```
+
+example:
+```yaml
+# SuperBench rules
+version: v0.3
+superbench:
+  rules:
+    default:
+      function: value
+      criteria: '>,0'
+      categories: Failed
+      metrics:
+        - kernel-launch/return_code
+        - mem-bw/return_code
+        - nccl-bw/return_code
+        - ib-loopback/return_code
+    rule0:
```

Comment:
Do we think rule0, rule1, rule2 are good? Or we need give a name. It's also good now.

Reply:
I think we can keep them now